pkg/scheduler/limits/definitions.go

```diff
@@ -8,3 +15,38 @@ type Limits interface {
 	// MaxQueryCapacity returns how much of the available query capacity can be used by this user.
 	MaxQueryCapacity(user string) float64
 }
+
+func QueueLimits(limits Limits) *queueLimits {
+	return &queueLimits{limits: limits}
+}
+
+type queueLimits struct {
+	limits Limits
+}
+
+// MaxConsumers is used to compute how many of the available consumers are allowed to handle requests for a given tenant.
```

Comment:
"consumers" is a term used by the tenant queue I think (correct me if I'm wrong), and I'm not sure it makes sense to use this language here. If we're talking about query workers, so let's refer to query workers.